### PR TITLE
ET-133 set staff role for lti

### PR DESCRIPTION
**Pull request by sruhela**
When users log in through LTI, recognize they are Staff and give them staff permissions.

### CODE REVIEW

File: lms/djangoapps/lti_provider/users.py [hunk 1]
```diff
@@ -92,12 +92,14 @@ def create_lti_user(lti_user_id, lti_consumer, lti_params=None):
...
                 except User.DoesNotExist:
                     pass
-
+   
                 edx_user = User.objects.create_user(
                     username=edx_username,
                     password=edx_password,
                     email=edx_email,
                 )
+                if 'Staff' in lti_role or 'Instructor' in lti_role or 'Administrator' in lti_role:
```
Comment (dmitry-viskov) [inline lms/djangoapps/lti_provider/users.py]: It is global permissions. As I understand you should use course permissions (see `CourseStaffRole`):
https://github.com/CredoReference/edx-platform/blob/integration-ficus/common/djangoapps/student/roles.py#L222

File: lms/djangoapps/lti_provider/views.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def lti_launch(request, course_id, usage_id):
     for key in lti_keys:
         if key in params:
             lti_params[lti_keys[key]] = params[key]
-    authenticate_lti_user(request, params['user_id'], lti_consumer, lti_params)
+    authenticate_lti_user(request, params['user_id'], params['roles'], lti_consumer, lti_params)
```
Comment (dmitry-viskov) [inline lms/djangoapps/lti_provider/views.py]: `roles` may not be transferred - you should predict this case

**Review by dmitry-viskov — CHANGES_REQUESTED** [review]
From the task description:
> Make sure instructor responses are excluded from reports (required for IMS certification)

i don't understand this condition. What reports are referring to? Today all reports include all answers from all users (students / instructors / admins / etc)

**Comment by dpowell** [comment]
We will (in a future enchantment) want an option in Insights to exclude results from staff and instructors but the first step here is when launched via LTI as an instructor create a user with that role on the course team so that they get the instructor view of our course.

File: lms/djangoapps/lti_provider/users.py (hunk 1 above)
Comment (dpowell) [reply lms/djangoapps/lti_provider/users.py]: Agreed.

File: lms/djangoapps/lti_provider/views.py (hunk 2 above)
Comment (sruhela) [reply lms/djangoapps/lti_provider/views.py]: Right now roles is a required parameter for the launch and if the role is unrecognized we just allow them to log in as a learner. Should that be different?

Comment (dpowell) [reply lms/djangoapps/lti_provider/views.py]: Agree, no role or an unrecognized role gets you a normal (student) user.

I think @dmitry-Viskov's concern was that if the `roles` key is not present in params, it'll blow up. You can use `params.get('roles')` instead, which will return `None` if the key is not found.

File: lms/djangoapps/lti_provider/views.py [hunk 3]
```diff
@@ -181,3 +185,11 @@ def parse_course_and_usage_keys(course_id, usage_id):
     usage_id = unquote_slashes(usage_id)
     usage_key = UsageKey.from_string(usage_id).map_into_course(course_key)
     return course_key, usage_key
+
+
+def set_user_roles(edx_user, course_key, roles):
+    global_admin = AdminFactory()
+    if 'Staff' in roles or 'Administrator' in roles:
```
Comment (dmitry-viskov) [inline lms/djangoapps/lti_provider/views.py]: Looks too complicated. I may suggest that such statement will be enough:
```
CourseStaffRole(course_key).add_users(user)
```

File: lms/djangoapps/lti_provider/views.py (hunk 2 above)
Comment (dmitry-viskov) [reply lms/djangoapps/lti_provider/views.py]: Yes. I mean `params.get('roles')`

File: lms/djangoapps/lti_provider/views.py [hunk 4]
```diff
@@ -181,3 +185,11 @@ def parse_course_and_usage_keys(course_id, usage_id):
...
+        auth.add_users(global_admin, CourseStaffRole(course_key), edx_user)
+    if 'Instructor' in roles or 'Administrator' in roles:
```
Comment (dmitry-viskov) [inline lms/djangoapps/lti_provider/views.py]: 'Instructor' is Course Admin
'Staff' is Course Staff

May be I'm wrong but we need to clarify this question. What roles are allowed. I may suggest that only 'Staff' (that is Course Staff)

Comment (sruhela) [reply lms/djangoapps/lti_provider/views.py]: 'Instructor' is the only one required for the certification, so I just left 'staff' and 'instructor' for now.

**Review by dmitry-viskov — APPROVED** [review]

**Pull request merged**